------------------------------------------------------------------------
r107018 | vboxsync | 2024-11-15 20:03:37 +0100 (Fri, 15 Nov 2024) | 1 line

3D: Fix for trivial index out-of-bound bug sometimes causing a crash during VM startup. bugref:10580

------------------------------------------------------------------------
Index: src/VBox/Devices/Graphics/DevVGA-SVGA.cpp
===================================================================
--- a/src/VBox/Devices/Graphics/DevVGA-SVGA.cpp	(revision 107017)
+++ b/src/VBox/Devices/Graphics/DevVGA-SVGA.cpp	(revision 107018)
@@ -7082,7 +7082,7 @@
 
     if (!pThis->svga.fVMSVGA2dGBO)
     {
-        for (unsigned i = 0; i <= RT_ELEMENTS(pThis->svga.au32DevCaps); ++i)
+        for (unsigned i = 0; i < RT_ELEMENTS(pThis->svga.au32DevCaps); ++i)
         {
             uint32_t val = 0;
             int rc = vmsvga3dQueryCaps(pThisCC, (SVGA3dDevCapIndex)i, &val);
